documents/entityrelationship/model.go

```diff
@@ -0,0 +1,319 @@
+}
+
+// loadFromP2PProtobuf loads the Entity Relationship from Centrifuge protobuf EntityRelationshipData.
+func (e *EntityRelationship) loadFromP2PProtobuf(entityRelationship *entitypb.EntityRelationship) error {
+	if entityRelationship.OwnerIdentity != nil {
+		did, err := identity.NewDIDFromBytes(entityRelationship.OwnerIdentity)
+		if err != nil {
+			return err
+		}
+		e.OwnerIdentity = &did
+	}
+
+	if entityRelationship.TargetIdentity != nil {
+		tid, err := identity.NewDIDFromBytes(entityRelationship.TargetIdentity)
+		if err != nil {
+			return err
+		}
+		e.TargetIdentity = &tid
+	}
+
+	e.Label = entityRelationship.Label
+
+	return nil
+}
+
+// PackCoreDocument packs the EntityRelationship into a CoreDocument.
+func (e *EntityRelationship) PackCoreDocument() (cd coredocumentpb.CoreDocument, err error) {
+	entityRelationship := e.createP2PProtobuf()
+	data, err := proto.Marshal(entityRelationship)
+	if err != nil {
+		return cd, errors.New("couldn't serialise EntityData: %v", err)
+	}
+
+	embedData := &any.Any{
+		TypeUrl: e.DocumentType(),
+		Value:   data,
+	}
+
+	return e.CoreDocument.PackCoreDocument(embedData), nil
+}
+
+// UnpackCoreDocument unpacks the core document into an EntityRelationship.
+func (e *EntityRelationship) UnpackCoreDocument(cd coredocumentpb.CoreDocument) error {
+	if cd.EmbeddedData == nil ||
+		cd.EmbeddedData.TypeUrl != e.DocumentType() {
+		return errors.New("trying to convert document with incorrect schema")
+	}
+
+	entityRelationship := new(entitypb.EntityRelationship)
+	err := proto.Unmarshal(cd.EmbeddedData.Value, entityRelationship)
+	if err != nil {
+		return err
+	}
+
+	err = e.loadFromP2PProtobuf(entityRelationship)
+	if err != nil {
+		return err
+	}
+	e.CoreDocument = documents.NewCoreDocumentFromProtobuf(cd)
+	return nil
+}
+
+// JSON marshals EntityRelationship into a json bytes
+func (e *EntityRelationship) JSON() ([]byte, error) {
+	return json.Marshal(e)
+}
+
+// FromJSON unmarshals the json bytes into EntityRelationship
+func (e *EntityRelationship) FromJSON(jsonData []byte) error {
+	return json.Unmarshal(jsonData, e)
+}
+
+// Type gives the EntityRelationship type.
+func (e *EntityRelationship) Type() reflect.Type {
+	return reflect.TypeOf(e)
+}
+
+// CalculateDataRoot calculates the data root.
+func (e *EntityRelationship) CalculateDataRoot() ([]byte, error) {
+	t, err := e.getDocumentDataTree()
+	if err != nil {
+		return nil, errors.New("failed to get data tree: %v", err)
+	}
+
+	dr := t.RootHash()
+	return dr, nil
+}
+
+// getDocumentDataTree creates precise-proofs data tree for the model
+func (e *EntityRelationship) getDocumentDataTree() (tree *proofs.DocumentTree, err error) {
+	eProto := e.createP2PProtobuf()
+	if err != nil {
+		return nil, err
+	}
+	if e.CoreDocument == nil {
+		return nil, errors.New("getDocumentDataTree error CoreDocument not set")
+	}
+	t := e.CoreDocument.DefaultTreeWithPrefix(prefix, compactPrefix())
+	err = t.AddLeavesFromDocument(eProto)
+	if err != nil {
```

Comment:
```suggestion
	if err := t.AddLeavesFromDocument(eProto); err != nil {
```

same for the next block as well